spec/ameba/rule/style/query_bool_methods_spec.cr

```diff
@@ -0,0 +1,60 @@
+require "../../../spec_helper"
+
+module Ameba::Rule::Style
+  subject = QueryBoolMethods.new
+
+  describe QueryBoolMethods do
+    it "passes for valid cases" do
+      expect_no_issues subject, <<-CRYSTAL
+        class Foo
+          class_property? foo = true
+          property? foo = true
+          property foo2 : Bool? = true
+          setter panda = true
+        end
+
+        module Bar
+          class_getter? bar : Bool = true
+          getter? bar : Bool
+          getter bar2 : Bool? = true
+          setter panda : Bool = true
+
+          def initialize(@bar = true)
+          end
+        end
+        CRYSTAL
+    end
+
+    {% for call in %w[getter class_getter property class_property] %}
+      it "reports `{{ call.id }}` assign with Bool" do
+        expect_issue subject, <<-CRYSTAL, call: {{ call }}
+          class Foo
+            %{call}   foo = true
+            _{call} # ^^^ error: Consider using '%{call}?' for 'foo'
+          end
+          CRYSTAL
+      end
+
+      it "reports `{{ call.id }}` type declaration assign with Bool" do
+        expect_issue subject, <<-CRYSTAL, call: {{ call }}
+          class Foo
+            %{call}   foo : Bool = true
+            _{call} # ^ error: Consider using '%{call}?' for 'foo'
+          end
+          CRYSTAL
+      end
+
+      it "reports `{{ call.id }}` type declaration with Bool" do
+        expect_issue subject, <<-CRYSTAL, call: {{ call }}
+          class Foo
+            %{call}   foo : Bool
+            _{call} # ^ error: Consider using '%{call}?' for 'foo'
+
+            def initialize(@foo = true)
+            end
+          end
+          CRYSTAL
+      end
+    {% end %}
```

Comment:
please add a test that checks if there are two boolean properties and only one is invalid

Reply:
I don't think I understand what do you mean.

Reply:
I still don't get it... Adjacent properties do not affect each other...

Reply:
> Adjacent properties do not affect each other

That actually highly depends on the implementation. Here is an example:

```diff
diff --git a/src/ameba/rule/style/query_bool_methods.cr b/src/ameba/rule/style/query_bool_methods.cr
index 4bd8c370..0750b143 100644
--- a/src/ameba/rule/style/query_bool_methods.cr
+++ b/src/ameba/rule/style/query_bool_methods.cr
@@ -66,6 +66,7 @@ module Ameba::Rule::Style
             issue_for name_node, MSG % {exp.name, name_node}
           end
         end
+        return
       end
     end
   end
```

if i slightly change the code, the test above fails, but all other pass